### ISSUE TITLE
Create "Permission Denied" page

### ISSUE THREAD

**Issue by marykatefain** [HarvestHub/GardenHub#105]
I recommend creating a "Permission Denied" page that the user gets redirected to anytime the test_func() fails. This could be a simple page like "Sorry, you do not have permission to access this page. Please contact your Garden Manager or a Garden Hub Admin if you feel this a mistake."

**Comment by alexgleason** [comment]
I think we should render a "permission denied" template instead of redirecting to a new page. I think it's better practice to display the error on the page being requested instead of redirecting. It seems more semantically relevant to tightly couple a displayed error with a request.

**Comment by JenniferGold** [comment]
Should all the references, including in error messages, to GardenHub be
without the space between Garden and Hub?

Thoughts re: the space between?

**Comment by alexgleason** [comment]
@JenniferGold I think "GardenHub" (without a space) insinuates that we are referring to the name of a specific product as opposed to describing a product using a generic name. GardenHub *is* a "garden hub" but I don't think "Garden Hub" is very friendly as a proper noun. That's my 2 cents.

**Comment by marykatefain** [comment]
I agree, Jen. Should be one word - that was just a typo on my part earlier!

**Comment by JenniferGold** [comment]
So we all agree no space?

Or did I misinterpret your comment, Alex?

**Comment by alexgleason** [comment]
Agree, no space
On Jan 25, 2018 10:44 PM, JenniferGold <notifications@github.com> wrote:So we all agree no space?

Or did I misinterpret your comment, Alex?

**Comment by alexgleason** [comment]
This is a bit complicated because it needs to distinguish between a user who can legitimately access the page in question but is not currently logged into an account which can, versus a user who is poking around and trying to access pages they know they shouldn't be able to see.

I'm not that concerned about what we do to handle the latter, because it is a case of a tech savvy user who is intentionally trying to subvert the system rather than using it as intended. I'm fine confusing someone in that scenario.

Directing users towards pages they can't access is a mistake and we should focus on conditionally hiding those options (I'm aware of none which still exist). Having this page would be a nice fallback in case we make a mistake, but I'd rather focus on testing the views so we don't make a mistake.

Actually implementing this would require rendering the "forbidden" template based on whether the user passes test_func *and* whether or not the user is currently logged in; a logged out user should *always* see a login screen because we can't assume to know their identity. A logged in user who fails test_func is the only condition where we may want to render an error.

Of course, we might still want to give the user a way to switch accounts from the forbidden screen, since someone could be trying to access a page they can legitimately access except using a different account.

I say we put it on the backburner for now.
